### PR TITLE
Write doc meta files

### PR DESCRIPTION
**Pull request by pclausen**
This is not my proudest moment of Python code, but it works and that was my main objective.

bin/create_fortran_test.py now creates:

exercises\practice\<exercise>\<exercise>_test.f90
exercises\practice\<exercise>\<exercise>.f90
exercises\practice\<exercise>\.meta\tests.toml
exercises\practice\<exercise>\.docs\instructions.md
exercises\practice\<exercise>\.meta\config.json

Example output
```
python bin\create_fortran_test.py -
j ..\problem-specifications\exercises\triangle\canonical-data.json -t exercises\practice\triangle\triangle_test.f90
Namespace(json='..\\problem-specifications\\exercises\\triangle\\canonical-data.json', target='exercises\\practice\\triangle\\triangle_test.f90')
Wrote : exercises\practice\triangle\triangle_test.f90
wrote stub : exercises\practice\triangle\triangle.f90
wrote exercises\practice\triangle\.meta\tests.toml
wrote exercises\practice\triangle\.docs\instructions.md
wrote exercises\practice\triangle\.meta\config.json
```

### CODE REVIEW

File: bin/create_fortran_test.py [hunk 1]
```diff
@@ -80,22 +81,27 @@ def write_testcase(c, tnum):
...
+    #print('A', TEST_NUMBER)
+    for c in j['cases']:
         if 'cases' in c:
             si.extend(create_single_test(c))
         else:
-            si.extend(write_testcase(c, tnum))
+            #print('B', TEST_NUMBER)
+            TEST_NUMBER = TEST_NUMBER +1
+            si.extend(write_testcase(c,TEST_NUMBER))
     return si
```
Comment (hiljusti) [inline bin/create_fortran_test.py:105]: To do this a little more cleanly we can flatten the test cases before handling them.

You're already flattening the test cases through recursion, and using the call stack as a data structure for managing it. You can convert the from using the call stack to using an actual stack

```python
si = []

nested_cases = j['cases']
flattened_cases = []

while nested_cases.len() > 0:
    cases = nested_cases.pop()
    if 'cases' in cases:
        map(nested_cases.append, cases['cases'])
    else:
        map(flattened_cases.append, cases)
        
for test_num, case in enumerate(flattened_cases):
    # etc
```

That's one option. Maybe better would be to have a function `flatten_test_cases` or something. (It also be flattened recursively of course)

File: bin/create_fortran_test.py [hunk 2]
```diff
@@ -155,25 +161,131 @@ def create_test(test_name, json_name):
...
     create 
     
         .meta/tests.toml 
         .meta/config.json
         .docs/instructions.md  (based on ../problem-specifications/exercises/[EXERCISE]/description.md)
 
-    for test_name
+    for test_file_name
 
     Not yet implemented.
```
Comment (hiljusti) [inline bin/create_fortran_test.py:175]: Is this still true?

File: bin/create_fortran_test.py [hunk 3]
```diff
@@ -155,25 +161,131 @@ def create_test(test_name, json_name):
...
+    write_tests_toml(json_name, test_toml)
+
     doc_dir = os.path.join( test_dir_name, '.docs')
+    desc_file = os.path.join( os.path.dirname(json_name),  'description.md')
+    instruction_file =  os.path.join(doc_dir, 'instructions.md')
+    write_instructions(desc_file, instruction_file)
+    
+    meta_yaml = os.path.join( os.path.dirname(json_name),  'metadata.yml')
+    local_config_json = os.path.join(meta_dir, 'config.json')
+    write_config_json(exercise_name, meta_yaml, local_config_json)
```
Comment (hiljusti) [inline bin/create_fortran_test.py:190]: The code could have a little better separation of concerns. This function is handling only filepaths, and the `write_etc` functions are doing both `read` and `write`

I suggest to either pull the `read` logic up to this level, or to create extra functions for the read logic that return either `str` or `list(str)`

File: bin/create_fortran_test.py (hunk 1 above)
Comment (pclausen) [reply bin/create_fortran_test.py]: Very good idea, did just that :-)

File: bin/create_fortran_test.py (hunk 3 above)
Comment (pclausen) [reply bin/create_fortran_test.py]: Tried to do this in commit 240739e. Not quite sure if it is much more readable.

**Comment by pclausen** [comment]
@hiljusti I reworked the script a bit. I got the feeling during it that I should really create a test for this script but put it on the TODO list for another day...

**Comment by hiljusti** [comment]
> @hiljusti I reworked the script a bit. I got the feeling during it that I should really create a test for this script but put it on the TODO list for another day...

Sounds good - I'm fine with "It works" for a single script

**Review by hiljusti — APPROVED** [review]

**Pull request merged**